diff --git a/android/src/main/java/expo/modules/intentlauncher/IntentLauncherModule.kt b/android/src/main/java/expo/modules/intentlauncher/IntentLauncherModule.kt
index 2a85454974457eb6bd01714687959b2cad08697c..19ee1c6c9f22393539192aaaf2e1d9720cd7c44f 100644
--- a/android/src/main/java/expo/modules/intentlauncher/IntentLauncherModule.kt
+++ b/android/src/main/java/expo/modules/intentlauncher/IntentLauncherModule.kt
@@ -40,6 +40,7 @@ class IntentLauncherModule : Module() {
         throw ActivityAlreadyStartedException()
       }
       val intent = Intent(activityAction)
+      params.packageName?.let { intent.setPackage(it) }
 
       params.className?.let {
         intent.component =
